packages/sitecore-jss-react/src/components/SitecoreContext.test.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { ComponentFactory } from './sharedTypes';
+import { expect } from 'chai';
+// const iterate = require('leakage').iterate;
+import { SitecoreContext, SitecoreContextFactory } from './SitecoreContext';
+import { withSitecoreContext } from '../enhancers/withSitecoreContext';
+const create = require('react-test-renderer').create;
+
+const TestNestedComponent: React.FC = (props: any) => (<div>{props.sitecoreContext && 'test'}</div>);
+const TestNestedComponentWithContext: React.FC = withSitecoreContext()(TestNestedComponent as any);
+const arrayOfComponents: React.Component[] = [];
+const mockComponentFactory: ComponentFactory = (name: string) => arrayOfComponents[name as any];
+const singleton = new SitecoreContextFactory();
```

Comment:
@M1r1k Please, refactor it a little bit in this way:
You can use `ComponentConsumerProps` when you are using `withSitecoreContext`;
```
import React, { FC } from 'react';
import { SitecoreContext, SitecoreContextFactory } from './SitecoreContext';
import { withSitecoreContext, ComponentConsumerProps } from '../enhancers/withSitecoreContext';

....

const NestedComponent: FC<ComponentConsumerProps> = (props: ComponentConsumerProps) => <div>{props.sitecoreContext && 'test'}</div>;
const NestedComponentWithContext: FC = withSitecoreContext()(NestedComponent);

const components = new Map();
const mockComponentFactory: ComponentFactory = name => components.get(name);

const singleton = new SitecoreContextFactory();
```